#### Multi-omics integration and regulatory inference for unpaired single-cell data with a graph-linked unified embedding framework - Cao and Gao 2021

https://github.com/gao-lab/GLUE 

One major obstacle of unpaired multi-omics integration is the feature  discrepancies among omics layers

GLUE (graph-linked unified embedding) = utilizes  accessible prior knowledge about regulatory interactions to bridge the  gaps between feature spaces. 

Systematic benchmarks demonstrated that  GLUE is accurate, robust and scalable. We further employed GLUE for  various challenging tasks, including triple-omics integration,  model-based regulatory inference and multi-omics human cell atlas  construction (over millions of cells) and found that GLUE achieved  superior performance for each task. As a generalizable framework, GLUE  features a modular design that can be flexibly extended and enhanced for new analysis tasks. The full package is available online at https://github.com/gao-lab/GLUE for the community.

Computationally, one major obstacle faced when integrating  unpaired multi-omics data is the distinct feature spaces of different  modalities (e.g., accessible chromatin regions in scATAC-seq vs. genes  in scRNA-seq)[14](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-14). A quick fix is to convert multimodality data into one common feature  space based on prior information and apply single-omics data integration methods[15](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-15)-[17](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-17). Such explicit “feature conversion” is straightforward, but has been reported to result in significant information loss[18](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-18). Algorithms based on coupled matrix factorization circumvent explicit conversion but hardly handle more than two omics layers[19](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-19), [20](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-20). An alternative option is to match cells from different omics layers via nonlinear manifold alignment, which removes the requirement of prior  knowledge completely and could reduce inter-modality information loss in theory[21](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-21), [22](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-22); however, this technique has mostly been applied to continuous, trajectory-like manifolds rather than atlases.



Hereby, we introduce GLUE (graph-linked unified embedding), a modular framework for integrating unpaired single-cell multi-omics data and inferring regulatory interactions simultaneously. By modeling the  regulatory interactions across omics layers explicitly, GLUE bridges the gaps between various omics-specific feature spaces in a biologically  intuitive manner. Systematic benchmarks and case studies demonstrate  that GLUE is accurate, robust and scalable for heterogeneous single-cell multi-omics data. Furthermore, GLUE is designed as a generalizable  framework that allows for easy extension and quick adoption to  particular scenarios in a modular manner. GLUE is publicly accessible at https://github.com/gao-lab/GLUE.

### Integrating unpaired single-cell multi-omics data via graph-guided embeddings

Inspired by previous works, we model cell states as low-dimensional cell embeddings learned through variational autoencoders[27](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-27), [28](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-28). Given their intrinsic differences in biological nature and assay  technology, each omics layer is equipped with a separate autoencoder  that uses a probabilistic generative model tailored to the  layer-specific feature space ([Fig. 1](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#F1), Methods).

Taking advantage of prior biological knowledge, we propose  the use of a knowledge-based graph (“guidance graph”) that explicitly  models cross-layer regulatory interactions for linking layer-specific  feature spaces; the vertices in the graph correspond to the features of  different omics layers, and edges represent signed regulatory  interactions. For example, when integrating scRNA-seq and scATAC-seq  data, the vertices are genes and accessible chromatin regions (i.e.,  ATAC peaks), and a positive edge can be connected between an accessible  region and its putative downstream gene. Then, adversarial multimodal  alignment is performed as an iterative optimization procedure, guided by feature embeddings encoded from the graph[29](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-29) ([Fig. 1](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#F1), Methods). Notably, when the iterative process converges, the graph can  be refined with inputs from the alignment procedure and used for  data-oriented regulatory inference (see below for more details).

### GLUE enables effective triple-omics integration

Unlike chromatin accessibility, gene body DNA methylation  generally shows a negative correlation with gene expression in neuronal  cells[36](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-36). GLUE natively supports the mixture of regulatory effects by modeling  edge signs in the guidance graph. Such a strategy avoids data inversion, which is required by previous methods[16](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-16), [17](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-17) and can break data sparsity and the underlying distribution. For the  triple-omics guidance graph, we linked gene body mCH and mCG levels to  genes via negative edges, while the positive edges between accessible  regions and genes remained the same.







### Model-based regulatory inference with GLUE

The  incorporation of a graph explicitly modeling regulatory interactions in  GLUE further enables a Bayesian-like approach that combines prior  knowledge and observed data for posterior regulatory inference.  Specifically, since the feature embeddings are designed to reconstruct  the knowledge-based guidance graph and single-cell multi-omics data  simultaneously ([Fig. 1](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#F1)), their cosine similarities should reflect information from both aspects, which we adopt as “regulatory scores”.



o capture remote cis-regulatory interactions, we employed a long-range  guidance graph connecting ATAC peaks and RNA genes within 150 kb windows weighted by a power-law function that models chromatin contact  probability[39](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-39)



The GLUE framework also allows additional regulatory evidence, such as  pcHi-C, to be incorporated intuitively via the guidance graph. Thus, we  further trained new models with a composite guidance graph containing  distance-weighted interactions as well as pcHi-C- and eQTL-supported  interactions (Supplementary Fig. 9)



the GLUE-inferred cis-regulatory interactions could provide new hints about the regulatory mechanisms of known TF-target pairs. 



Combining omics-specific autoencoders with graph-based  coupling and adversarial alignment, we designed and implemented the GLUE framework for unpaired single-cell multi-omics data integration with  superior accuracy and robustness. By modeling regulatory interactions  across omics layers explicitly, GLUE uniquely supports model-based  regulatory inference for unpaired multi-omics datasets, exhibiting even  higher reliability than regular correlation analysis on paired datasets  (notably, in a Bayesian interpretation, the GLUE regulatory inference  can be seen as a posterior estimate, which can be continuously refined  upon the arrival of new data). Furthermore, benefitting from a neural  network-based design, GLUE enables notable scalability for whole-atlas  alignment over millions of unpaired cells, which remains a serious  challenge for *in silico* integration. In fact, we also attempted to perform integration using online iNMF, which was the only other  method capable of integrating the data at full scale, but the result was far from optimal (Supplementary Fig. 18a, b). Meanwhile, an attempt to  integrate the data as aggregated metacells (Methods) via the popular  Seurat v3 method also failed (Supplementary Fig. 18c, d).

Unpaired multi-omics integration, also referred to as diagonal integration[14](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-14), shares some conceptual similarities with batch effect correction[46](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-46), as both call for the alignment of unpaired cells in certain data  representations. Nonetheless, the former is significantly more  challenging because of the distinct, omics-specific feature spaces.  While completely unsupervised multi-omics integration has been proposed[21](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-21), [22](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-22), such an approach is exceedingly difficult and has largely been limited  to aligning continuous trajectories. For general-case multi-omics  integration, additional prior knowledge is necessary. At the omics  feature level, presumed feature interactions have been used via feature  conversion[15](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-15)-[17](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-17), [30](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-30) or coupled matrix factorization[19](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-19), [20](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-20). While feature conversion may seem to be a straightforward solution, the inevitable information loss[18](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-18) can have a detrimental effect on performance. Apart from the feature-converted data, Seurat v3[15](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-15) and bindSC[30](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-30) also devised heuristic strategies to utilize information in the  original feature space, which probably explains their improved  performance than methods that do not[16](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-16), [17](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-17). At the cell level, known cell types have also been used via (semi-)supervised learning[47](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-47), [48](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-48), but this approach incurs substantial limitations in terms of  applicability since such supervision is typically unavailable and in  many cases serves as the purpose of multi-omics integration *per se*[26](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-26). Notably, one of these methods was proposed with a similar autoencoder architecture and adversarial alignment[48](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-48), but it relied on matched cell types or clusters to orient the  alignment. In fact, GLUE shares more conceptual similarity with the  coupled matrix factorization methods, but with superior accuracy,  robustness and scalability, which mostly benefits from its deep  generative model-based design.

We note that the current  framework also works for integrating omics layers with shared features,  by using either the same vertex or connected surrogate vertices for each shared feature in the guidance graph. In particular, the integration  between scRNA-seq and spatial transcriptomics[49](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-49), [50](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-50) could be naturally implemented in this way. After the integration,  genes not detected in the spatial transcriptome could be further imputed via cross-layer translation, through a combination of the spatial  transcriptomics encoder and the scRNA-seq decoder.

As a  generalizable framework, GLUE features a modular design, where the data  and graph autoencoders are independently configurable.

- The data autoencoders in GLUE are customizable with appropriate generative  models that conform to omics-specific data distributions. In the current work, we used the negative binomial distribution for scRNA-seq and  scATAC-seq, and the zero-inflated log-normal distribution for snmC-seq  (Methods). Nevertheless, generative distributions can be easily  reconfigured to accommodate other omics layers, such as protein  abundance[51](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-51) and histone modification[52](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-52), and to adopt new advances in data modeling techniques[53](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-53).
- The guidance graphs used in GLUE have currently been limited to  multipartite graphs, containing only edges between features of different layers. Nonetheless, graphs, as intuitive and flexible representations  of regulatory knowledge, can embody more complex regulatory patterns,  including within-modality interactions, non-feature vertices, and  multi-relations. Beyond canonical graph convolution, more advanced graph neural network architectures[54](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-54)-[56](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-56) may also be adopted to extract richer information from the regulatory graph.

Recent advances in experimental multi-omics technologies have increased the availability of paired data[8](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-8)-[11](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-11), [31](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-31). While most of the current simultaneous multi-omics protocols still  suffer from lower data quality or throughput than that of single-omics  methods[57](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-57), paired cells can be highly informative in anchoring different omics  layers and should be utilized in conjunction with unpaired cells  whenever available. It is straightforward to extend the GLUE framework  to incorporate such pairing information, e.g., by adding another loss  term that penalizes the embedding distances between paired cells[58](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-58). Such an extension may ultimately lead to a solution for the general case of mosaic integration[14](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-14).

Apart from multi-omics integration, we also note that the GLUE framework  could be suitable for cross-species integration, especially when distal  species are concerned and one-to-one orthologs are limited.  Specifically, we may compile all orthologs into a GLUE guidance graph  and perform integration without explicit ortholog conversion. Under that setting, the GLUE approach could also be conceptually connected to a  recent work called SAMap[59](https://www.biorxiv.org/content/10.1101/2021.08.22.457275v2.full#ref-59).

We believe that GLUE, as a modular and generalizable framework, creates an unprecedented opportunity towards effectively delineating gene  regulatory maps via large-scale multi-omics integration at single-cell  resolution. The whole package of GLUE, along with tutorials and demo  cases, is available online at https://github.com/gao-lab/GLUE for the community.